testing/test_mark.py

```diff
@@ -22,6 +22,19 @@ def test_pytest_mark_notcallable(self):
         mark = Mark()
         pytest.raises((AttributeError, TypeError), mark)
 
+    def test_mark_with_param(self):
+        def some_function(abc):
+            pass
+
+        class SomeClass(object):
+            pass
+
+        assert pytest.mark.fun(some_function) is some_function
+        assert pytest.mark.fun.with_args(some_function) is not some_function
```

Comment:
> i merely want to assert return value behavior, the test as is is sufficient and minimal

Until someone refactors the code and this starts to return not `some_function` but also not a `MarkInfo`, and then unexpected breakage occurs. 😜 

But OK then.